testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java

```diff
@@ -0,0 +1,90 @@
+import io.trino.tests.product.launcher.env.common.HydraIdentityProvider;
+import io.trino.tests.product.launcher.env.common.SeleniumChrome;
+import io.trino.tests.product.launcher.env.common.Standard;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_CONFIG_PROPERTIES;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+@TestsEnvironment
+public class EnvSinglenodeOauth2HttpProxy
+        extends EnvironmentProvider
+{
+    private final PortBinder binder;
+    private final HydraIdentityProvider hydraIdentityProvider;
+    private final ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeOauth2HttpProxy(DockerFiles dockerFiles, PortBinder binder, Standard standard, HydraIdentityProvider hydraIdentityProvider, SeleniumChrome seleniumChrome)
+    {
+        super(ImmutableList.of(standard, hydraIdentityProvider, seleniumChrome));
+
+        this.binder = requireNonNull(binder, "binder is null");
+        this.hydraIdentityProvider = requireNonNull(hydraIdentityProvider, "hydraIdentityProvider is null");
+        requireNonNull(dockerFiles, "dockerFiles is null");
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-oauth2-http-proxy/");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(COORDINATOR, dockerContainer -> {
+            dockerContainer
+                    .withCopyFileToContainer(
+                            forHostPath(configDir.getPath("config.properties")),
+                            CONTAINER_PRESTO_CONFIG_PROPERTIES);
+
+            binder.exposePort(dockerContainer, 7778);
+        });
+
+        DockerContainer hydraClientConfig = hydraIdentityProvider.createClient(
+                builder,
+                "trinodb_client_id",
+                "trinodb_client_secret",
+                "client_secret_basic",
+                "trinodb_client_id/",
+                "https://presto-master:7778/oauth2/callback,https://localhost:7778/oauth2/callback");
+
+        builder.containerDependsOn(COORDINATOR, hydraClientConfig.getLogicalName());
+
+        DockerContainer proxy = new DockerContainer("httpd:2.4.51", "proxy");
+        proxy.withCopyFileToContainer(forHostPath(configDir.getPath("httpd.conf")), "/usr/local/apache2/conf/httpd.conf");
+        builder.addContainer(proxy);
+        builder.containerDependsOn(COORDINATOR, "proxy");
+    }
+
+    @Override
+    public List<EnvironmentExtender> getDependencies()
+    {
+        return ImmutableList.of(hydraIdentityProvider);
```

Comment:
Is this required when `hydraIdentityProvider` is already in `super(ImmutableList.of(standard, hydraIdentityProvider, seleniumChrome));`?